src/main/java/duke/Deadline.java

```diff
@@ -0,0 +1,58 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+/**
+ * This class is a subclass of Task.
+ * @author Sim Jun Heng
+ * @version CS2103T AY21/22 Sem 2
+ */
+public class Deadline extends Task {
+
+    private LocalDate date;
+
+    /**
+     * Assigns desc, date and done to this instance.
+     *
+     * @param desc the task description.
+     * @param date the date of the event.
+     * @param isDone the current completion status of the task.
+     */
+    public Deadline(String desc, LocalDate date, boolean isDone) {
+        super(desc, isDone);
+        this.date = date;
+    }
+
+    @Override
+    public String toString() {
+        String str = "[D]";
+        // Convert Current Date Format to MMM dd yyyy
+        String customPattern = "MMM dd yyyy";
+        DateTimeFormatter customFormat = DateTimeFormatter.ofPattern(customPattern);
+        String tempDate = customFormat.format(this.date);
+        if (super.isDone) {
+            str += "[X] " + super.desc + " (by: " + tempDate + ")";
+        } else {
+            str += "[ ] " + super.desc + " (by: " + tempDate + ")";
+        }
```

Comment:
In my opinion, I prefer to write strings in java using template literals, such as 
```
String.format("[X] %s by: %s)", super.desc, tempDate);
```
This way, it is not as easy to miss out a " " (space), which could lead to bugs that could take hours to resolve. If this syntax interests you, you could try this next time :)

Reply:
I agree with this. 

In fact, I created a custom `OutputFormatter` just for a bulk `append` method for `StringBuilder`.